server/core/basic_cluster.go

```diff
@@ -25,21 +27,48 @@ import (
 	"go.uber.org/zap"
 )
 
+// SelectConfig provides parameters for selecting regions.
+type SelectConfig struct {
+	NewRegionFirst bool
+	NewProbability float64
+	MaxRegionCount int
+	TimeThreshold  uint64
+}
+
 // BasicCluster provides basic data member and interface for a tikv cluster.
 type BasicCluster struct {
 	sync.RWMutex
-	Stores  *StoresInfo
-	Regions *RegionsInfo
+	Stores     *StoresInfo
+	Regions    *RegionsInfo
+	NewRegions *regionCache
+	SelectConf *SelectConfig
 }
 
 // NewBasicCluster creates a BasicCluster.
 func NewBasicCluster() *BasicCluster {
 	return &BasicCluster{
-		Stores:  NewStoresInfo(),
-		Regions: NewRegionsInfo(),
+		Stores:     NewStoresInfo(),
+		Regions:    NewRegionsInfo(),
+		NewRegions: newRegionCache(),
+		SelectConf: NewSelectConfig(),
```

Comment:
IMO, it's kind of weird to add a config here.

Reply:
This optimization is for balancing region. Perhaps config can be managed by balance_region scheduler. When scheduling, config is passed to cluster as a parameter. Is this modification feasible?